// Copyright (C) 2020 Ellen Poe
//
// This file is part of MASS.
//
// MASS is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// MASS is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with MASS.  If not, see <http://www.gnu.org/licenses/>.

syntax = "proto3";
package api;

import "mass/api/actions.proto";
import "mass/api/updates.proto";
import "mass/api/scenario.proto";

service MassBackend {
  rpc Connect(stream ConnectRequest) returns (stream VesselUpdate);
  rpc DoAction(stream DoActionRequest) returns (DoActionResponse);
}

message ConnectRequest {
  // The unique ID that the user wants to connect to.
  string vessel_unique_id = 1;
  // The scenario to instantiate.
  Scenario scenario = 2;
  // The name of the scenario instance. To connect to an existing scenario
  // instance, populate this field but leave scenario empty.
  string scenario_id = 3;
}
